t/change_extension.t

```diff
@@ -0,0 +1,47 @@
+use 5.008001;
+use strict;
+use warnings;
+use Test::More 0.96;
+
+use lib 't/lib';
+
+use Path::Tiny;
+
+
+{
+    my $renamed1 = path("C:/mydir/myfile.com.extension")->change_extension(".old");
+    ok($renamed1->stringify eq 'C:/mydir/myfile.com.old', 'rename to .old with leading perdiod');
```

Comment:
I strongly prefer [table-driven tests](https://www.perl.com/article/178/2015/6/17/Separate-data-and-behavior-with-table-driven-testing/) because it's much easier to extend with new cases.  See `t/subsumes.t` for an example as well.  Here, you've grouped tests inside brackets, which is similar to how tests are grouped in `subsumes.t`.